artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerTarget.java

```diff
@@ -264,6 +265,14 @@ public void flow() {
    protected void actualDelivery(Message message, Delivery delivery, DeliveryAnnotations deliveryAnnotations, Receiver receiver, Transaction tx) {
       OperationContext oldContext = recoverContext();
       incrementSettle();
+      boolean noForward = false;
+      if (receiver.getRemoteDesiredCapabilities() != null) {
+         for (Symbol capability : receiver.getRemoteDesiredCapabilities()) {
+            if (capability == AMQPMirrorControllerSource.NO_FORWARD) {
+               noForward = true;
+            }
+         }
+      }
```

Comment:
This isnt going to change for the duration of the link, so it makes sense to evaluate it once at the start when initializing the mirror target, rather than determining the same result repeatedly for each arriving message.

Reply:
That makes a lot of sense. Thanks.